frontend/src/store.js

```diff
@@ -21,16 +21,97 @@ export default new Vuex.Store({
     ...nativeLanguageMutations
   },
   actions: {
+    ...nativeLanguageActions,
+
     async getAllStudents({ commit }) {
       const response = await fetch("/api/students")
       const data = await response.json()
       commit(MUTATIONS.SET_STUDENTS, data)
     },
-    ...nativeLanguageActions
 
+    // Update student status
+    async updateStudentStatus({ commit, dispatch }, { studentID, previousStatusString, nextStatusString, updatedBy }) {
+      var success = true;
+
+      // PATCH student
+      const studentRequestOptions = {
+        method: "PATCH",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({StudentID: studentID, StatusString: nextStatusString})
+      }
+
+      fetch("/api/students/" + studentID, studentRequestOptions)
+        .then(response => {
+            // If PATCH fails, return
+            if(response.status !== 200) {
+              success = false;
+              return;
+            }
+          }
+        )
+
+      // POST statusUpdate
+      const statusUpdateRequestOptions = {
+        method: "POST",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          StudentID: studentID,
+          PreviousStatusString: previousStatusString,
+          NextStatusString: nextStatusString,
+          UpdatedBy: updatedBy
+        })
+      }
+      fetch("/api/statusUpdates", statusUpdateRequestOptions)
+        .then(response => {
+            // If PATCH fails, return
+            if(response.status !== 200) {
+              success = false;
+              return;
+            }
+          }
+        )
+
+      // Check if both PATCH student and POST statusUpdate succeed
+      if (success) { dispatch('getAllStudents') }
```

Comment:
Just making sure that we are aware of the asynchronous flow here. Each `fetch` is asynchronous, so the second fetch is going to start potentially before the first one is done. Similarly, the last line checking whether it was successful could be executed before both fetches are done (and hence we don't know if they were successful).

This is the "async structure" of your code:
```js
fetch("/api/students") // 1. Make first fetch
  .then(response => {
    // 2. do stuff after the fetch is done
  })

fetch("/api/statusUpdates") // 3. Makes second fetch
  .then(response => {
    // 4. do stuff after the fetch is done
  })

// 5. do something at the end of everything (e.g. `getAllStudents`).
```
Let's say the fetch is slow. The order of execution is going to be: 1, 3, 5, 2, 4. If you want it to be 1, 2, 3, 4, 5, you will need a structure more like:
```js
fetch("/api/students")
  .then(response => {
    // first fetch has finished, do what you need
  })
  .then(() => {
    return fetch("api/statusUpdates")
  })
  .then(response => {
    // first and second fetch have finished, do what you need
  })
  .then(() => {
    // do something here, knowing both have finished.
  })
```

Reply:
Alternatively, if you don't want the second fetch to "wait" for the first one, but collectively you want to wait for both, you can do:
```js
const firstFetch = fetch("/api/students").then(response => ...)
const secondFetch = fetch("/api/students").then(response => ...)

Promise.all([
  firstFetch,
  secondFetch
])
  .then(() => { /* Success */ })
  .catch(() => { /* Something went wrong */ })
```
The 2 fetches will be started, but the `Promise.all` gives us a way to wait for both of them before moving on to the next action (which if it is successful, should be the dispatch to refresh the students list).

Reply:
Thanks Thomas! We have worked in the async fetch checks according to your suggestion. Good catch!